plugins/woocommerce/includes/abstracts/abstract-wc-data.php

```diff
@@ -506,6 +506,26 @@ public function delete_meta_data( $key ) {
 		}
 	}
 
+	/**
+	 * Delete meta data with a matching value.
+	 *
+	 * @since 7.7.0
+	 * @param string $key   Meta key.
+	 * @param mixed  $value Meta value. Entries will only be removed that match the value.
+	 */
+	public function delete_matched_meta_data( $key, $value ) {
```

Comment:
Optional suggestion: `delete_meta_data_value` seems like a more meaningful / descriptive method name.

Reply:
I thought about `delete_meta_data_with_value` since that seems even more semantically correct, but it also seemed too verbose 🤔 

Yeah, `delete_meta_data_value` seems good.

Reply:
4f38c05